### PR TITLE
Add gdrive for go-storage design

### PR DESCRIPTION
**Pull request by junaire**
In this PR, we will discuss some details about our design.

### CODE REVIEW

File: docs/rfcs/14-gdrive-for-go-storage-design.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+## Implementation
+
+When users try to call `Write("foo/bar/test.txt")`, first we use `pathToId` to try to convert the path to `FileID`. If it fails, then it means that this file doesn't exist, and we should create one. Then we can create folder `foo` , `bar`, and last the file `test.txt`. 
+
+Our significant point `pathToId` can be implement like this:
+
+If the file is in the root folder, then we just do a simple search by using `drive.service.Files.List().Q(searchArgs).Do()`. The return value type is `*drive.File`, and it's attribute `ID` is what we need.
+
+But if the file path is like `foo/bar/demo.txt`, it would be a little complex.
```
Comment (Xuanwo) [inline docs/rfcs/14-gdrive-for-go-storage-design.md:44]: If the path is complex like `a/b/c/d/e/f/h/q.txt`, it looks like we need to repeat the search many times.

How about cache the `path -> id` in memory with TTL?

Comment (Xuanwo) [reply docs/rfcs/14-gdrive-for-go-storage-design.md:44]: ping @junaire for a look.

Comment (junaire) [reply docs/rfcs/14-gdrive-for-go-storage-design.md:44]: OK, I'll work on it. However, I don't have too much experience with this and I would like to ask if we should use some third library like [go-cache](https://github.com/patrickmn/go-cache)? Or just simply use the map from the standard library?

Comment (Xuanwo) [reply docs/rfcs/14-gdrive-for-go-storage-design.md:44]: To simplify the implementation, use a map with Mutex is OK for now.

We can implement the TTL logic later by https://github.com/dgraph-io/ristretto or other libs.

File: docs/rfcs/0-example.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+- Author: (fill me in with `name <mail>`, e.g., Xuanwo <github@xuanwo.io>)
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: [beyondstorage/go-storage#0](https://github.com/beyondstorage/go-storage/issues/0)
```
Comment (Xuanwo) [inline docs/rfcs/0-example.md]:
```suggestion
- RFC PR: [beyondstorage/go-service-gdrive#0](https://github.com/beyondstorage/go-service-gdrive/issues/0)
```

File: docs/rfcs/0-example.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+- Tracking Issue: [beyondstorage/go-storage#0](https://github.com/beyondstorage/go-storage/issues/0)
```
Comment (Xuanwo) [inline docs/rfcs/0-example.md]:
```suggestion
- Tracking Issue: [beyondstorage/go-service-gdrive#0](https://github.com/beyondstorage/go-service-gdrive/issues/0)
```

File: docs/rfcs/0-example.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+# GSP-0: <proposal name>
```
Comment (Xuanwo) [inline docs/rfcs/0-example.md]:
```suggestion
# RFC-0: <proposal name>
```

File: docs/rfcs/0-example.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+- Updates: (delete this part if not applicable)
+  - [GSP-20](./20-abc): Deletes something
```
Comment (Xuanwo) [inline docs/rfcs/0-example.md]:
```suggestion
  - [RFC-20](./20-abc): Deletes something
```

File: docs/rfcs/0-example.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+- Updated By: (delete this part if not applicable)
+  - [GSP-10](./10-do-be-do-be-do): Adds something
```
Comment (Xuanwo) [inline docs/rfcs/0-example.md]:
```suggestion
  - [RFC-10](./10-do-be-do-be-do): Adds something
```

File: docs/rfcs/0-example.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  - [GSP-1000](./1000-lalala): Deprecates this RFC
```
Comment (Xuanwo) [inline docs/rfcs/0-example.md]:
```suggestion
  - [RFC-1000](./1000-lalala): Deprecates this RFC
```

File: docs/rfcs/14-gdrive-for-go-storage-design.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
+- Author: Jun jun@junz.org
+- Start Date: 2021-7-18
+- RFC PR: [beyondstorage/go-service-gdrive#14](https://github.com/beyondstorage/go-service-gdrive/issues/14)
+- Tracking Issue: [beyondstorage/go-service-gdrive#15](https://github.com/beyondstorage/go-service-gdrive/issues/15)
+
+# GSP-14: Gdrive for go-storage design
```
Comment (Xuanwo) [inline docs/rfcs/14-gdrive-for-go-storage-design.md]:
```suggestion
# RFC-14: Gdrive for go-storage design
```

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**